### ISSUE TITLE
Operator crashes at startup in native build

### ISSUE THREAD

**Issue by sclorng** [quarkiverse/quarkus-operator-sdk#175]
I'm migrating from 1.9.x to 2.0.x and when runing in native, it crashes at startup:

```
2021-12-18 13:40:13,153 INFO  traceId= [io.jav.ope.Operator] (main) Operator SDK 1.10.1 (commit: f7aceb4) built on Fri Dec 10 13:52:03 GMT 2021 starting...
2021-12-18 13:40:13,153 INFO  traceId= [io.jav.ope.Operator] (main) Client version: 5.10.1
2021-12-18 13:40:13,350 ERROR traceId= [io.jav.ope.Operator] (main) Error retrieving the server version: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:314)
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:266)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:569)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:527)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:510)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.restCall(OperationSupport.java:782)
        at io.fabric8.kubernetes.client.BaseClient.getVersionInfo(BaseClient.java:174)
        at io.fabric8.kubernetes.client.BaseKubernetesClient.getVersion(BaseKubernetesClient.java:494)
        at io.fabric8.openshift.client.DefaultOpenShiftClient.getVersion(DefaultOpenShiftClient.java:535)
        at io.javaoperatorsdk.operator.Operator.start(Operator.java:74)
        at foo.bar.QuarkusOperator.run(QuarkusOperator.java:29)
        at foo.bar.QuarkusOperator_ClientProxy.run(Unknown Source)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:125)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at foo.bar.QuarkusOperator.main(QuarkusOperator.java:23)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.fabric8.kubernetes.client.VersionInfo` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (BufferedInputStream); line: 2, column: 3]
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1764)
        at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
        at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1209)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1415)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:362)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:195)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4593)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3593)
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:312)
        ... 15 more

2021-12-18 13:40:13,351 ERROR traceId= [io.qua.run.Application] (main) Failed to start application (with profile prod): com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.fabric8.kubernetes.client.VersionInfo` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (BufferedInputStream); line: 2, column: 3]
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1764)
        at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
        at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1209)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1415)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:362)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:195)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4593)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3593)
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:312)
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:266)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:569)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:527)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:510)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.restCall(OperationSupport.java:782)
        at io.fabric8.kubernetes.client.BaseClient.getVersionInfo(BaseClient.java:174)
        at io.fabric8.kubernetes.client.BaseKubernetesClient.getVersion(BaseKubernetesClient.java:494)
        at io.fabric8.openshift.client.DefaultOpenShiftClient.getVersion(DefaultOpenShiftClient.java:535)
        at io.javaoperatorsdk.operator.Operator.start(Operator.java:74)
        at foo.bar.QuarkusOperator.run(QuarkusOperator.java:29)
        at foo.bar.QuarkusOperator_ClientProxy.run(Unknown Source)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:125)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at foo.bar.QuarkusOperator.main(QuarkusOperator.java:23)
```

The VersionInfo class seems not to be registered for reflection. I guess it should be done in the quarkus-kubernetes extension ?

**Comment by metacosm** [comment]
Does this also happen in non-native mode?

**Comment by scrocquesel** [comment]
No

**Comment by scrocquesel** [comment]
I don't know when the fix will be release in quarkus. Shall we add `@RegisterForReflection(targets = {VersionInfo.class})` somewhere for the time being ?

**Comment by gsmet** [comment]
It will be included in 2.6.1.Final that I will release Friday if all goes well.

**Comment by metacosm** [comment]
I'm starting to think we should remove the version output from the SDK as it's not the first time it's causing issues… 😅

**Comment by scrocquesel** [comment]
Does the it test run in native mode ?

**Comment by metacosm** [comment]
No, but they definitely should. I haven't had time to look into it… Is it something you could look into? Not sure if we can run native tests on GitHub… 😞

**Comment by gsmet** [comment]
You sure can. We do it for Quarkus.

**Comment by scrocquesel** [comment]
@metacosm I tried adding `-Dnative` on the build workflow but it doesn't compile on main branch :/

```
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]    classlist:   4,125.61 ms,  1.20 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]        (cap):     720.64 ms,  1.19 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]        setup:   3,285.26 ms,  1.19 GB
The bundle named: messages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
22:55:12,622 INFO  [org.jbo.threads] JBoss Threads version 3.4.2.Final
Warning: Could not register complete reflection metadata for io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl$3. Reason(s): java.lang.NoClassDefFoundError: org/apache/commons/codec/binary/Base64InputStream
Warning: Could not register complete reflection metadata for io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl$1. Reason(s): java.lang.NoClassDefFoundError: org/apache/commons/codec/binary/Base64InputStream
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]     (clinit):   1,928.64 ms,  4.33 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]   (typeflow):   4,376.21 ms,  4.33 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]    (objects):  99,083.67 ms,  4.33 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]   (features):  26,602.31 ms,  4.33 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]     analysis: 140,789.86 ms,  4.33 GB
[quarkus-operator-sdk-integration-tests-2.0.3-SNAPSHOT-runner:26]      [total]: 148,777.99 ms,  4.33 GB
Error: Unsupported features in 5 methods
Detailed message:
Error: Class initialization of io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl$1 failed. Use the option --initialize-at-run-time=io.fabric8.kubernetes.client.dsl.internal.core.v1.PodOperationsImpl$1 to explicitly request delayed initialization of this class.
Original exception that caused the problem: java.lang.NoClassDefFoundError: org/apache/commons/codec/binary/Base64InputStream
```

Tried with 2.0.2 tag but got same errors. I'll check tomorrow at work which version I was using or if I have any custom config.

**Comment by metacosm** [comment]
@scrocquesel working on this actually

**Comment by metacosm** [comment]
This should now be fixed with the upgrade to Fabric8 5.11.1.
